src/sidebar/util/fetch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * An error indicating a failed network request.
+ *
+ * Failures that this error can represent include:
+ *
+ *  - Failures to send an HTTP request
+ *  - Requests that returned non-2xx responses
+ *  - Failures to parse the response in the expected format (eg. JSON)
+ */
+export class FetchError extends Error {
+  /**
+   * @param {Response|null} response - The response to the `fetch` request or
+   *   `null` if the fetch failed
+   * @param {string} [reason] - Additional details about the error. This might
+   *   include context of the network request or a server-provided error in
+   *   the response.
+   */
+  constructor(response, reason = '') {
+    let message = 'Network request failed';
```

Comment:
Capturing the `URL` would be essential.

Reply:
It is not only 'Network` errors here. The network can be fine and fail the URL parameters, parsing response, etc.

Reply:
I guess one way to indirectly get the URL is by looking at the stacktrace.

Reply:
Adding the request URL to the error seems like a good idea, so I've done that.

Reply:
Have you considered using the `cause` property like you did in `TokenError`?

`constructor(url, response, cause) {`